kubernetes/resource_kubernetes_certificate_signing_request.go

```diff
@@ -0,0 +1,196 @@
+								"3. Otherwise, it is assigned `kubernetes.io/legacy-unknown`. Distribution of trust for signers happens out of band." +
+								"You can select on this field using `spec.signerName`.",
+							Optional: true,
+							ForceNew:    true,
+						},
+						"usages": {
+							Type:        schema.TypeSet,
+							Description: apiDocSpec["usages"],
+							Set:         schema.HashString,
+							Elem:        &schema.Schema{Type: schema.TypeString},
+							Optional:    true,
+							ForceNew:    true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceKubernetesCertificateSigningRequestCreate(d *schema.ResourceData, meta interface{}) error {
+	conn, err := meta.(KubeClientsets).MainClientset()
+	if err != nil {
+		return err
+	}
+
+	metadata := expandMetadata(d.Get("metadata").([]interface{}))
+	spec, err := expandCertificateSigningRequestSpec(d.Get("spec").([]interface{}))
+	if err != nil {
+		return err
+	}
+
+	csr := v1beta1.CertificateSigningRequest{
+		ObjectMeta: metadata,
+		Spec:       *spec,
+	}
+	log.Printf("[INFO] Creating new certificate signing request: %#v", csr)
+	_, err = conn.CertificatesV1beta1().CertificateSigningRequests().Create(&csr)
+	if err != nil {
+		return fmt.Errorf("Failed to create certificate signing request: %s", err)
+	}
+
+	if d.Get("auto_approve").(bool) {
+        retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			pendingCSR, getErr := conn.CertificatesV1beta1().CertificateSigningRequests().Get(metadata.Name, metav1.GetOptions{})
+			if getErr != nil {
+				return getErr
+			}
+			approval := v1beta1.CertificateSigningRequestCondition{
+				Type: "Approved",
+				Reason: "TerraformAutoApprove",
+				Message: "This CSR was approved by Terraform auto_approve.",
+			}
+			pendingCSR.Status.Certificate = []byte{}
+			pendingCSR.Status.Conditions = append(pendingCSR.Status.Conditions, approval)
+			_, updateErr := conn.CertificatesV1beta1().CertificateSigningRequests().UpdateApproval(pendingCSR)
+			return updateErr
+		})
+	if retryErr != nil {
+		panic(fmt.Errorf("CSR auto-approve update failed: %v", retryErr))
+	}
+	fmt.Println("CSR auto-approve update succeeded")
+		if err != nil {
+			return err
+		}
+	}
+
+	log.Printf("[DEBUG] Waiting for certificate to be issued")
+	stateConf := &resource.StateChangeConf{
+			Target:  []string{"Approved"},
+			Pending: []string{""},
+			Timeout: d.Timeout(schema.TimeoutCreate),
+			Refresh: func() (interface{}, string, error) {
+				out, refreshErr := conn.CertificatesV1beta1().CertificateSigningRequests().Get(metadata.Name, metav1.GetOptions{})
+				if refreshErr != nil {
+					log.Printf("[ERROR] Received error: %#v", refreshErr)
+					return out, "Error", err
+				}
+				statusConditions := ""
+				emptyStatus := v1beta1.CertificateSigningRequestStatus{}
+
+				if !reflect.DeepEqual(emptyStatus, out.Status) {
+					statusConditions = fmt.Sprintf("%v", out.Status.Conditions[0].Type)
```

Comment:
I want to see if I can search for the condition in this array instead. WIP

Reply:
You should be able to do a linear search on the array using the `type` field.

Reply:
Oops, I just saw this comment. Let me know if you want me to solve this a different way. The code snippet there is outdated.